pkg/webhook/client.go

```diff
@@ -108,8 +109,17 @@ func (c *Client[Req, Res]) Send(
 			return resp.StatusCode, nil
 		}
 
-		if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
-			return resp.StatusCode, ErrUnexpectedResponse
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return resp.StatusCode, fmt.Errorf("read response body: %w", err)
+		}
+
+		if err := json.Unmarshal(bodyBytes, &res); err != nil {
+			logger := logging.From(ctx)
+			if logger != nil {
+				logger.Errorf("webhook json decode failed: status=%d, body=%s", resp.StatusCode, string(bodyBytes))
+			}
+			return resp.StatusCode, fmt.Errorf("%w: invalid json response", ErrUnexpectedResponse)
```

Comment:
```suggestion
			return resp.StatusCode, fmt.Errorf("invalid json response: %w", ErrUnexpectedResponse)
```

https://github.com/uber-go/guide/blob/master/style.md#error-wrapping